backend/telemetry_core/src/aggregator/inner_loop.rs

```diff
@@ -386,10 +386,11 @@ impl InnerLoop {
                 let node_id = match self.node_ids.remove_by_right(&(shard_conn_id, local_id)) {
                     Some((node_id, _)) => node_id,
                     None => {
-                        log::error!(
-                            "Cannot find ID for node with shard/connectionId of {:?}/{:?}",
-                            shard_conn_id,
-                            local_id
+                        // It's possible that some race between removing and disconnecting shards might lead to
+                        // more than one remove message for the same node. This isn't really a problem, but we
+                        // hope it won't happen so make a note if it does:
+                        log::debug!(
```

Comment:
Are we worried about the process consuming too much memory because we may never call `remove_nodes_and_broadcast_result` here? Leading to a potentially OOM kill? 

Would it be possible for a malicious party to force the shard to submit `Remove { random_id }`?
And in that could we "mute" / disconnect the shard if it makes N wrong `Remove` calls?

Reply:
It shouldn't be possible for any party to have any control over any of the ids; we assign them all internally. 

And if the id doesn't exist for this remove message, it means the node has been removed already :)

Remove calls are also entirely internal; an external person cannot cause it to be called except by disconnecting (or when messages with certain ids go stale).